src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -5394,6 +5417,253 @@ def delete_deployment(self, deployment_id: UUID) -> None:
+        """
+        count = session.scalar(
+            select(func.count())
+            .select_from(ArtifactVisualizationSchema)
+            .where(
+                ArtifactVisualizationSchema.artifact_version_id
+                == artifact_version_id
+            )
+        )
+        if not count or visualization_index >= count:
+            raise IllegalOperationError(
+                "Artifact version "
+                f"`{artifact_version_id}` does not expose a visualization "
+                f"with index {visualization_index}."
+            )
+
+    def _assert_curated_visualization_duplicate(
+        self,
+        session: Session,
+        *,
+        artifact_version_id: UUID,
+        visualization_index: int,
+        resource_id: UUID,
+        resource_type: VisualizationResourceTypes,
+    ) -> None:
+        """Ensure a curated visualization link does not already exist.
+
+        Args:
+            session: The session to use.
+            artifact_version_id: The ID of the artifact version to validate.
+            visualization_index: The index of the visualization to validate.
+            resource_id: The ID of the resource to validate.
+            resource_type: The type of the resource to validate.
+
+        Raises:
+            EntityExistsError: If a curated visualization link already exists.
+        """
+        existing = session.exec(
+            select(CuratedVisualizationSchema)
+            .where(
+                CuratedVisualizationSchema.artifact_version_id
+                == artifact_version_id
+            )
+            .where(
+                CuratedVisualizationSchema.visualization_index
+                == visualization_index
+            )
+            .where(CuratedVisualizationSchema.resource_id == resource_id)
+            .where(
+                CuratedVisualizationSchema.resource_type == resource_type.value
+            )
+        ).first()
+        if existing is not None:
+            raise EntityExistsError(
+                "A curated visualization for this resource already exists "
+                "for the specified artifact version and visualization index."
+            )
+
+    def create_curated_visualization(
+        self, visualization: CuratedVisualizationRequest
+    ) -> CuratedVisualizationResponse:
+        """Persist a curated visualization link.
+
+        Args:
+            visualization: The curated visualization to persist.
+
+        Returns:
+            The persisted curated visualization.
+
+        Raises:
+            IllegalOperationError: If the resource does not belong to the same project as the curated visualization.
+            KeyError: If the resource does not exist.
+        """
+        with Session(self.engine) as session:
+            self._set_request_user_id(
+                request_model=visualization, session=session
+            )
+
+            artifact_version: ArtifactVersionSchema = self._get_schema_by_id(
```

Comment:
Does the `_get_reference_schema_by_id` method that we usually use to check the project consistency not work here?